src/pyconcrete/__init__.py

```diff
@@ -16,52 +16,29 @@
 
 import imp
 import marshal
+import struct
 import sys
-from os.path import exists, isdir, join
+from importlib._bootstrap_external import _get_supported_file_loaders
+from importlib.machinery import FileFinder, SourceFileLoader, SOURCE_SUFFIXES
+from . import _pyconcrete  # noqa: E402
+
 
-EXT_PY = '.py'
-EXT_PYC = '.pyc'
-EXT_PYD = '.pyd'
 EXT_PYE = '.pye'
 
 __all__ = ["info"]
 
-from . import _pyconcrete  # noqa: E402
 
 info = _pyconcrete.info
 encrypt_file = _pyconcrete.encrypt_file
 decrypt_file = _pyconcrete.decrypt_file
 decrypt_buffer = _pyconcrete.decrypt_buffer
 
+SOURCE_SUFFIXES.append(EXT_PYE)
```

Comment:
Does it is a best practice to modify the list of `SOURCE_SUFFIXES`? Can we just clone the suffixes in the internal function?

Reply:
We need to modify it, because it used in `all_suffixes()` function and called by `inspect.py` in
```
def getmodulename(path):
    """Return the module name for a given file, or None."""
    fname = os.path.basename(path)
    # Check for paths that look like an actual module file
    suffixes = [(-len(suffix), suffix)
                    for suffix in importlib.machinery.all_suffixes()]
    suffixes.sort() # try longest suffixes first, in case they overlap
    for neglen, suffix in suffixes:
        if fname.endswith(suffix):
            return fname[:neglen]
    return None
```
and `getmodulename` is used in `pkgutil.py` by Django
This looks like a hack, but the second point is that we need to be able to detect the module name relative to `.pye` files, because `.py` can be deleted by us

Reply:
[I added commit with comment](https://github.com/Falldog/pyconcrete/pull/108/commits/0d2537f40b5ba11ed46f542dc5cbf5e0115f4d10)

Reply:
Thanks for your elaboration. It's a great improvement.